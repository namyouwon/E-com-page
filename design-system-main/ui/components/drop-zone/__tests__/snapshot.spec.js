// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license
/* eslint-env jest */

import React from 'react';
import createHelpers from '../../../../jest.helpers';
import {
  DropZone,
  DropZoneInsertionPoint,
  DropZoneContainer,
  DropZoneLabel
} from '../base/example';

const { matchesMarkup } = createHelpers(__dirname);

describe('Drop Zone Base', () => {
  it('renders the default drop zone', () => matchesMarkup(<DropZone />));
  it('renders drop zone when dragging', () =>
    matchesMarkup(<DropZone isDragging />));
  it('renders the insertion point', () =>
    matchesMarkup(<DropZoneInsertionPoint />));
  it('renders the container', () =>
    matchesMarkup(<DropZoneContainer hasActions />));
  it('renders container when hovered', () =>
    matchesMarkup(<DropZoneContainer isHovered />));
  it('renders container with no actions', () =>
    matchesMarkup(<DropZoneContainer />));
  it('renders the label', () =>
    matchesMarkup(<DropZoneLabel headerText="Content" />));
  it('renders the container label', () =>
    matchesMarkup(<DropZoneLabel isContainer headerText="Component Name" />));
});
