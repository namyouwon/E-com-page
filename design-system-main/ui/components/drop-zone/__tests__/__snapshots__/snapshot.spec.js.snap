// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drop Zone Base renders container when hovered 1`] = `
<div
  className="slds-drop-zone__container slds-is-hovered"
  tabIndex="-1"
>
  <div
    className="slds-drop-zone__label slds-drop-zone__label_container"
  >
    <div
      className="slds-media slds-media_center"
    >
      <div
        className="slds-media__figure"
      >
        <span
          className="slds-icon_container slds-icon-utility-connected_apps slds-current-color"
          title={null}
        >
          <svg
            aria-hidden={true}
            className="slds-icon slds-icon slds-icon_x-small slds-icon-text-default"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#connected_apps"
            />
          </svg>
        </span>
      </div>
      <div
        className="slds-media__body"
      >
        <h2>
          <button
            className="slds-drop-zone__label_button slds-button_reset"
          >
            <span
              className="slds-assistive-text"
            >
              Edit: 
            </span>
            <span>
              Component Name
            </span>
          </button>
        </h2>
      </div>
    </div>
  </div>
</div>
`;

exports[`Drop Zone Base renders container with no actions 1`] = `
<div
  className="slds-drop-zone__container"
  tabIndex="-1"
>
  <div
    className="slds-drop-zone__label slds-drop-zone__label_container"
  >
    <div
      className="slds-media slds-media_center"
    >
      <div
        className="slds-media__figure"
      >
        <span
          className="slds-icon_container slds-icon-utility-connected_apps slds-current-color"
          title={null}
        >
          <svg
            aria-hidden={true}
            className="slds-icon slds-icon slds-icon_x-small slds-icon-text-default"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#connected_apps"
            />
          </svg>
        </span>
      </div>
      <div
        className="slds-media__body"
      >
        <h2>
          <button
            className="slds-drop-zone__label_button slds-button_reset"
          >
            <span
              className="slds-assistive-text"
            >
              Edit: 
            </span>
            <span>
              Component Name
            </span>
          </button>
        </h2>
      </div>
    </div>
  </div>
</div>
`;

exports[`Drop Zone Base renders drop zone when dragging 1`] = `
<div
  className="slds-drop-zone slds-drop-zone_drag"
/>
`;

exports[`Drop Zone Base renders the container 1`] = `
<div
  className="slds-drop-zone__container"
  tabIndex="-1"
>
  <div
    className="slds-drop-zone__label slds-drop-zone__label_container"
  >
    <div
      className="slds-media slds-media_center"
    >
      <div
        className="slds-media__figure"
      >
        <span
          className="slds-icon_container slds-icon-utility-connected_apps slds-current-color"
          title={null}
        >
          <svg
            aria-hidden={true}
            className="slds-icon slds-icon slds-icon_x-small slds-icon-text-default"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#connected_apps"
            />
          </svg>
        </span>
      </div>
      <div
        className="slds-media__body"
      >
        <h2>
          <button
            className="slds-drop-zone__label_button slds-button_reset"
          >
            <span
              className="slds-assistive-text"
            >
              Edit: 
            </span>
            <span>
              Component Name
            </span>
          </button>
        </h2>
      </div>
    </div>
  </div>
  <div
    className="slds-drop-zone__actions"
  >
    <div
      className="slds-button-group"
      role="group"
    >
      <button
        className="slds-button slds-button_icon slds-button_icon-brand slds-button_icon-x-small"
        onClick={[Function]}
        title="Move"
      >
        <svg
          aria-hidden={true}
          className="slds-button__icon"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#move"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          Move
        </span>
      </button>
      <button
        className="slds-button slds-button_icon slds-button_icon-brand slds-button_icon-x-small"
        onClick={[Function]}
        title="Close"
      >
        <svg
          aria-hidden={true}
          className="slds-button__icon"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          Close
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Drop Zone Base renders the container label 1`] = `
<div
  className="slds-drop-zone__label slds-drop-zone__label_container"
>
  <div
    className="slds-media slds-media_center"
  >
    <div
      className="slds-media__figure"
    >
      <span
        className="slds-icon_container slds-icon-utility-connected_apps slds-current-color"
        title={null}
      >
        <svg
          aria-hidden={true}
          className="slds-icon slds-icon slds-icon_x-small slds-icon-text-default"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#connected_apps"
          />
        </svg>
      </span>
    </div>
    <div
      className="slds-media__body"
    >
      <h2>
        <button
          className="slds-drop-zone__label_button slds-button_reset"
        >
          <span
            className="slds-assistive-text"
          >
            Edit: 
          </span>
          <span>
            Component Name
          </span>
        </button>
      </h2>
    </div>
  </div>
</div>
`;

exports[`Drop Zone Base renders the default drop zone 1`] = `
<div
  className="slds-drop-zone"
/>
`;

exports[`Drop Zone Base renders the insertion point 1`] = `
<div
  className="slds-drop-zone_drag__slot"
/>
`;

exports[`Drop Zone Base renders the label 1`] = `
<div
  className="slds-drop-zone__label"
>
  <div
    className="slds-media slds-media_center"
  >
    <div
      className="slds-media__figure"
    >
      <span
        className="slds-icon_container slds-icon-utility-builder slds-current-color"
        title={null}
      >
        <svg
          aria-hidden={true}
          className="slds-icon slds-icon slds-icon_x-small slds-icon-text-default"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#builder"
          />
        </svg>
      </span>
    </div>
    <div
      className="slds-media__body"
    >
      <h2>
        Content
      </h2>
    </div>
  </div>
</div>
`;
